src/main/haskell/kore/test/Test/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -506,168 +509,175 @@ test_andTermsSimplification = give mockSymbolOrAliasSorts
+            (Just Predicated
+                { term =
+                    Mock.concatMap
+                        (Mock.builtinMap [(Mock.aConcrete, fOfA)])
+                        (Mock.builtinMap [(Mock.bConcrete, fOfB)])
+                , predicate = makeTruePredicate
+                , substitution =
+                    [ (Mock.m, Mock.builtinMap [(Mock.bConcrete, fOfB)])
+                    , (Mock.x, fOfA)
+                    ]
+                }
+            )
+            (unify
+                mockMetadataTools
+                (Mock.concatMap
+                    (mkVar Mock.m)
+                    (Mock.builtinMap [(Mock.aConcrete, mkVar Mock.x)])
                 )
-        )
-    , testCase "builtin List domain"
-        (do
+                (Mock.builtinMap [(Mock.aConcrete, fOfA), (Mock.bConcrete, fOfB)])
+            )
+        ]
+    , testGroup "builtin List domain"
+        [
             let term1 = Mock.builtinList
                             [Mock.constr10 Mock.cf, Mock.constr11 Mock.cf]
-            assertEqualWithExplanation "[same head, same head]"
-                ( Just $ Predicated
-                        { term = term1
-                        , predicate = makeTruePredicate
-                        , substitution = []
-                        }
-                )
-                ( unify
-                    mockMetadataTools
-                        term1
-                        term1
-                )
-            let term3 = Mock.builtinList $
-                            [Mock.a, Mock.a]
-            let term4 = Mock.builtinList $
-                            [Mock.a, Mock.b]
-            assertEqualWithExplanation "[same head, different head]"
-                ( Just $ ExpandedPattern.bottom
-                )
-                ( unify
-                    mockMetadataTools
-                    term3
-                    term4
-                )
+            in
+                testCase "[same head, same head]"
+                $ assertEqualWithExplanation ""
+                    ( Just $ Predicated
+                            { term = term1
+                            , predicate = makeTruePredicate
+                            , substitution = []
+                            }
+                    )
+                    ( unify
+                        mockMetadataTools
+                            term1
+                            term1
+                    )
+        ,
+            let term3 = Mock.builtinList [Mock.a, Mock.a]
+                term4 = Mock.builtinList [Mock.a, Mock.b]
+                unified34 =
+                    let unifiedElements = [pure Mock.a, ExpandedPattern.bottom]
+                    in Mock.builtinList <$> sequenceA unifiedElements
+            in
+                testCase "[same head, different head]"
+                $ assertEqualWithExplanation ""
+                    (Just unified34)
```

Comment:
I think that the result should be just bottom, not a list containing bottom. They are equivalent, but the former is much more useful since it will probably transform the entire rewrite term to bottom. Why the change?

Reply:
The result is `\and(/* term: a list containing bottom */, /* predicate: */ \bottom)`, so the entire rewrite will be transformed into bottom. This is not immediately evident from the way the test is written, so I should change the way the test result is written, but is that result acceptable? I believe this is also what Map unification does. The change is because the List unification code is not doing its own ad-hoc unification now.

Reply:
Ah, `sequenceA` will `and` the predicate from ExpandedPattern.bottom with the one from Mock.a, producing a bottom predicate... Yes, this is acceptable. Yes, making this clear would be nice.